src/main/java/org/folio/inventory/dataimport/handlers/actions/ReplaceInstanceEventHandler.java

```diff
@@ -299,6 +300,10 @@ private Future<Void> prepareRecordForMapping(DataImportEventPayload dataImportEv
 
           if (instance.getSource().equals(MARC.getValue())) {
             incomingRecord.setMatchedId(existingRecord.getMatchedId());
+            if (nonNull(existingRecord.getGeneration())) {
+              int incrementedGeneration = existingRecord.getGeneration();
+              incomingRecord.setGeneration(++incrementedGeneration);
```

Comment:
propose to set record's generation to null so that the generation will be calculated in mod-source-record storage by the logic of the endpoint used in this handler for record update

Reply:
discussed with @Maksat-Galymzhan that during processing action to update instance for the case when a job profile contains action profiles for updating marc-bib and for updating instance the SRS logic for generation calculation doesn't account previous record's generation which was calculated during processing action for marc-bib update,